### PR TITLE
[WIP] ambient animals module

### PR DESCRIPTION
**Pull request by Krzyciu**
Added ambient animals module with dyn-sim support

### CODE REVIEW

File: addons/modules/modules/ambientAnimals/functions/fnc_module.sqf [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+
+// Exit if module executed inside editor, not on server or not in init mode
+if (is3DEN || !(isServer) || !(_mode isEqualTo "init")) exitWith {};
+_input params [
+  ["_logic", objNull, [objNull]],
+  ["_isActivated", false, [true]],
+  ["_isCuratorPlaced", false, [true]]
+];
+// Exit if module is null, not local or placed by zeus (should not happen)
+if (isNull _logic || !(local _logic) || _isCuratorPlaced) exitWith {};
```
Comment (SzwedzikPL) [inline addons/modules/modules/ambientAnimals/functions/fnc_module.sqf:17]: `isNull _logic || !(local _logic) || !_isActivated ||_isCuratorPlaced`

File: addons/modules/modules/ambientAnimals/functions/fnc_module.sqf [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+
+if (CBA_missionTime < 30) then {
+  sleep 15;
+};
+
+LOG_1('Starting execution of EXEC_MODULE_NAME (isActivated: %1).',str _isActivated);
+
+if (_isActivated) then {
```
Comment (SzwedzikPL) [inline addons/modules/modules/ambientAnimals/functions/fnc_module.sqf:25]: Remove this `if`

File: addons/modules/modules/ambientAnimals/functions/fnc_module.sqf [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+        LOG_3('Generating animals #%1 (animalClass: "%2").',str _animalIndex,_animalsType);
+
+        private _spawnPos = _logicArea call EFUNC(common,getAreaRandomPos);
+        private _animal = createAgent [_animalsType,[0,0,0],[],0,"NONE"];
+        _animal setPos _spawnPos;
+        _animal setDir (random 360);
+        _animal enableDynamicSimulation true;
+    };
+};
+
```
Comment (SzwedzikPL) [inline addons/modules/modules/ambientAnimals/functions/fnc_module.sqf:48]: Remove logic after execution - `deleteVehicle _logic`

File: addons/modules/modules/ambientAnimals/functions/fnc_module.sqf [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+if (is3DEN || !(isServer) || !(_mode isEqualTo "init")) exitWith {};
+_input params [
+  ["_logic", objNull, [objNull]],
+  ["_isActivated", false, [true]],
+  ["_isCuratorPlaced", false, [true]]
+];
+// Exit if module is null, not local or placed by zeus (should not happen)
+if (isNull _logic || !(local _logic) || _isCuratorPlaced) exitWith {};
+
+if (CBA_missionTime < 30) then {
```
Comment (SzwedzikPL) [inline addons/modules/modules/ambientAnimals/functions/fnc_module.sqf:19]: I know the reason why i used it in genSoldiers - compositions. Why it is used here?

File: addons/modules/modules/ambientAnimals/functions/fnc_module.sqf [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+  sleep 15;
+};
+
+LOG_1('Starting execution of EXEC_MODULE_NAME (isActivated: %1).',str _isActivated);
+
```
Comment (SzwedzikPL) [inline addons/modules/modules/ambientAnimals/functions/fnc_module.sqf:24]: Update module status for debug UI
```
if (is3DENPreview) then {
  [_logic, true] call EFUNC(debug,updateModule);
};
```

File: addons/modules/modules/ambientAnimals/functions/fnc_module.sqf [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+    _logicArea append (_logic getVariable ["objectarea", []]);
+
+    if ((_animalsType isEqualTo "") || (_animalsCount isEqualTo 0)) exitWith {
+      WARNING_2('EXEC_MODULE_NAME - animal type is empty or animal count is 0 (animalsType: %1, animalsCount: %2).',str _animalsType, _animalsCount);
+    };
+
+    // Generate animals
+    for "_animalIndex" from 0 to _animalsMaxIndex do {
+
+        LOG_3('Generating animals #%1 (animalClass: "%2").',str _animalIndex,_animalsType);
```
Comment (SzwedzikPL) [inline addons/modules/modules/ambientAnimals/functions/fnc_module.sqf:39]: I think one log before `for` will be enough. All generated animals are the same type/class (unlike gensoldiers), so no need to log it per spawn.

File: addons/modules/modules/ambientAnimals/functions/fnc_module.sqf [hunk 7]
```diff
@@ -0,0 +1,49 @@
...
+
+        private _spawnPos = _logicArea call EFUNC(common,getAreaRandomPos);
+        private _animal = createAgent [_animalsType,[0,0,0],[],0,"NONE"];
```
Comment (SzwedzikPL) [inline addons/modules/modules/ambientAnimals/functions/fnc_module.sqf:42]: `createAgent [_animalsType, [0,0,0], [], 0, "NONE"];`

File: addons/modules/modules/ambientAnimals/module.hpp [hunk 8]
```diff
@@ -0,0 +1,78 @@
...
+        };
+        class GVAR(animalsType): GVAR(dynamicCombo) {
+            displayName = CSTRING(ambientAnimals_animalsType);
+            tooltip = CSTRING(ambientAnimals_animalsType_Tooltip);
+            property = QGVAR(animalsType);
+            typeName = "STRING";
+            defaultValue = "'Rabbit_F'";
+            class values {
+                class Rabbit_F {
+                    name = CSTRING(ambientAnimals_Rabbit);
```
Comment (SzwedzikPL) [inline addons/modules/modules/ambientAnimals/module.hpp:31]: `CSTRING(ambientAnimals_animalsType_Rabbit)`

File: addons/modules/modules/ambientAnimals/module.hpp [hunk 9]
```diff
@@ -0,0 +1,78 @@
...
+                    tooltip = "";
+                    value = "Rabbit_F";
+                    default = 1;
+                };
+                class Hen_random_F {
+                    name = CSTRING(ambientAnimals_Hen);
```
Comment (SzwedzikPL) [inline addons/modules/modules/ambientAnimals/module.hpp:37]: `CSTRING(ambientAnimals_animalsType_Hen)`

File: addons/modules/modules/ambientAnimals/module.hpp [hunk 10]
```diff
@@ -0,0 +1,78 @@
...
+                    tooltip = "";
+                    value = "Hen_random_F";
+                };
+                class Fin_random_F {
+                    name = CSTRING(ambientAnimals_Dog);
```
Comment (SzwedzikPL) [inline addons/modules/modules/ambientAnimals/module.hpp:42]: `CSTRING(ambientAnimals_animalsType_Dog)`

File: addons/modules/modules/ambientAnimals/module.hpp [hunk 11]
```diff
@@ -0,0 +1,78 @@
...
+                    tooltip = "";
+                    value = "Fin_random_F";
+                };
+                class Goat_random_F {
+                    name = CSTRING(ambientAnimals_Goat);
```
Comment (SzwedzikPL) [inline addons/modules/modules/ambientAnimals/module.hpp:47]: `CSTRING(ambientAnimals_animalsType_Goat)`

File: addons/modules/modules/ambientAnimals/module.hpp [hunk 12]
```diff
@@ -0,0 +1,78 @@
...
+                    tooltip = "";
+                    value = "Goat_random_F";
+                };
+                class Sheep_random_F {
+                    name = CSTRING(ambientAnimals_Sheep);
```
Comment (SzwedzikPL) [inline addons/modules/modules/ambientAnimals/module.hpp:52]: `CSTRING(ambientAnimals_animalsType_Sheep)`

File: addons/modules/stringtable.xml [hunk 13]
```diff
@@ -1113,4 +1113,49 @@
       <Polish>Moduł do poprawnego działania musi być zsynchronizowany z dokładnie jedną (i tylko jedną) jednostką piechoty (pojazdy nie są obsługiwane) której grupa ma być użyta jako kompozycja.</Polish>
     </Key>
   </Package>
+</Package>
```
Comment (SzwedzikPL) [inline addons/modules/stringtable.xml:1116]: Fix indentation pls

**Review by SzwedzikPL — CHANGES_REQUESTED** [review]

File: addons/modules/modules/ambientAnimals/functions/fnc_module.sqf (hunk 4 above)
Comment (Krzyciu) [reply addons/modules/modules/ambientAnimals/functions/fnc_module.sqf:19]: Tested it on dedicated server, if more than 5 modules with max amount of animals were used at mission start, then animals count didn't match - 1 or 2 were skipped in each generator. Sleep resolved this issue.

**Pull request merged**